src/views/Viewer.vue

```diff
@@ -168,9 +168,6 @@ import '@nextcloud/dialogs/dist/index.css'
 import { showError } from '@nextcloud/dialogs'
 import { emit, subscribe, unsubscribe } from '@nextcloud/event-bus'
 
-import NcActionButton from '@nextcloud/vue/dist/Components/NcActionButton.js'
-import NcActionLink from '@nextcloud/vue/dist/Components/NcActionLink.js'
-import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
```

Comment:
Not fond of lazy loading components that are always included. We don't gain anything but a bit more weight and network usage :)

Reply:
![127 0 0 1_8888_](https://github.com/nextcloud/viewer/assets/14975046/47e24bb5-f067-4bbe-afa3-8b9301ace16b)

Reply:
> Since the bundle is prefetched and cached, the user can never know about the second request behind the scenes when the viewer is actually opened.

That is not the case Webpack does not preload the js, it will load them on demand.
If you're on slow network, it can adds up quite a lot :)

Example given: 1.46s to wait for the viewer to load on 4G.

Reply:
Yeah, you need to specify the `webpackPrefetch` and/or `webpackPreload`, to enable preloading ...

https://webpack.js.org/guides/code-splitting/#prefetchingpreloading-modules

Reply:
![image](https://github.com/nextcloud/viewer/assets/10709794/19d0fca9-8bea-42a0-b135-8e2f919f5a21)

Seems to work for me. If you enable `Disable Cache` in the JS console then it disables prefetching; could be related to that.

Reply:
> Seems to work for me. If you turn on `Disable Cache` in the JS console then it disables prefetching; could be related to that. The above is with a cold cache in incognito.

Nope even with cache disabled :thinking: 
Component, plyr or editor are loaded after the viewer is opened.

I'm also ok merging this, and we can always see later how it goes :shrug: 
But I kind of would like a more wider discussion on async components, so we can all decide to go the same direction?

Reply:
Would you be up to create a thread on https://github.com/nextcloud/standards/issues with your already impressive graphs and data to show the benefits/disadvantages and how-to of this?

Reply:
> Nope even with cache disabled 🤔

I meant it wont work with cache disabled 😅. But if it wasn't disabled earlier then definitely something is off 🤔 

> I'm also ok merging this, and we can always see later how it goes 🤷
> But I kind of would like a more wider discussion on async components, so we can all decide to go the same direction?

Sounds good 👍🏻 

> Would you be up to create a thread on https://github.com/nextcloud/standards/issues with your already impressive graphs and data to show the benefits/disadvantages and how-to of this?

Will do. Never knew about this repo. I think _most_ of the overhead is coming from `nextcloud-vue` at this point; desperately need tree shaking in there. If that can be fixed then lazy loading components would be pointless to begin with (happy dance).

Reply:
> Will do. Never knew about this repo. I think _most_ of the overhead is coming from `nextcloud-vue` at this point; desperately need tree shaking in there. If that can be fixed then lazy loading components would be pointless to begin with (happy dance).

This is on purpose, a bit more discrete to not become the place to discuss any thoughts. I think important topics like those should indeed be brought up there :+1: